plutus-benchmark/values/bench/Bench.hs

```diff
@@ -0,0 +1,239 @@
+            mkMockValues
+                mockNumPolicies1
+                mockNumTokensPerPolicy1
+                mockTokenValues1
+     in pure fValue
+
+setupEnvFlattened2 :: IO (FlattenedValue.Value, FlattenedValue.Value)
+setupEnvFlattened2 =
+    let (_, fValue1, _) =
+            mkMockValues
+                mockNumPolicies1
+                mockNumTokensPerPolicy1
+                mockTokenValues1
+        (_, fValue2, _) =
+            mkMockValues
+                mockNumPolicies2
+                mockNumTokensPerPolicy2
+                mockTokenValues2
+     in pure (fValue1, fValue2)
+
+setupEnvMM1 :: IO MMValue.Value
+setupEnvMM1 =
+    let (_, _, mmValue) =
+            mkMockValues
+                mockNumPolicies1
+                mockNumTokensPerPolicy1
+                mockTokenValues1
+     in pure mmValue
+
+setupEnvMM2 :: IO (MMValue.Value, MMValue.Value)
+setupEnvMM2 =
+    let (_, _, mmValue1) =
+            mkMockValues
+                mockNumPolicies1
+                mockNumTokensPerPolicy1
+                mockTokenValues1
+        (_, _, mmValue2) =
+            mkMockValues
+                mockNumPolicies2
+                mockNumTokensPerPolicy2
+                mockTokenValues2
+     in pure (mmValue1, mmValue2)
+
+setupEnvNested1andInv :: IO (NestedValue.Value, NestedValue.Value)
+setupEnvNested1andInv =
+    let (nValue1, _, _) =
+            mkMockValues
+                mockNumPolicies1
+                mockNumTokensPerPolicy1
+                mockTokenValues1
+        (nValue1Inv, _, _) =
+            mkMockValues
+                mockNumPolicies1
+                mockNumTokensPerPolicy1
+                (map negate mockTokenValues1)
+     in pure (nValue1, nValue1Inv)
+
+setupEnvFlattened1andInv :: IO (FlattenedValue.Value, FlattenedValue.Value)
+setupEnvFlattened1andInv =
+    let (_, fValue1, _) =
+            mkMockValues
+                mockNumPolicies1
+                mockNumTokensPerPolicy1
+                mockTokenValues1
+        (_, fValue1Inv, _) =
+            mkMockValues
+                mockNumPolicies1
+                mockNumTokensPerPolicy1
+                (map negate mockTokenValues1)
+     in pure (fValue1, fValue1Inv)
+
+setupEnvMM1andInv :: IO (MMValue.Value, MMValue.Value)
+setupEnvMM1andInv =
+    let (_, _, mmValue1) =
+            mkMockValues
+                mockNumPolicies1
+                mockNumTokensPerPolicy1
+                mockTokenValues1
+        (_, _, mmValue1Inv) =
+            mkMockValues
+                mockNumPolicies1
+                mockNumTokensPerPolicy1
+                (map negate mockTokenValues1)
+     in pure (mmValue1, mmValue1Inv)
+
+main :: IO ()
+main = do
+    config <- getConfig 30
+    defaultMainWith config $
+        [ env setupEnvNested1 $ \ ~nValue ->
+            bgroup "NestedValue"
+                [ bench "insertCoin - new coin"
+                    $ nf (NestedValue.insertCoin polId500 tokN10000 200) nValue
```

Comment:
You can't do this. `nf` is O(n) and `insertCoin` is what, O(log n)? The cost of your measurements is asymptotically higher than the cost of the operation whose cost you're trying to measure.